### PR TITLE
[feat-5266]: kto over iets anders feedback

### PR DESCRIPTION
**Pull request by vdegraaf**
Ticket: [SIG-5266](https://gemeente-amsterdam.atlassian.net/browse/SIG-5266)

- Correct order in history. Open answer should come after the related checkbox/radiobutton
- max 1000 characters in textareas

### CODE REVIEW

**Review by SireeKoolenWijkstra — COMMENTED** [review]
- [ ] why only the last letter that is being stripped? I've just entered a lot of dots, if you don't want them as the last letter, why would you want them as the second to last letter? 
- [ ] in the email a text of 1000 characters doesn't break but continues on the same line. 
- [ ] It is a little bit weird that you would use a inputfield for 1000 characters and an areafield for also 1000 characters (it's necessary because of the validation, but both of them on the same page look a  little weird)

File: src/signals/incident/containers/KtoContainer/utils.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ export const getMergedOpenAnswers = (object: FormData): FormData => {
   const objectWithoutOpenAnswers = {} as FormData
 
   Object.entries(object).forEach(([key, value]) => {
-    if (key.startsWith('open_answer-')) {
+    if (key.startsWith('open_answer-') && value) {
       const newKey = key.replace('open_answer-', '')
       openAnswerPairs[newKey] = value
```
[the remark this answers is not in the document]
Comment (vdegraaf) [reply src/signals/incident/containers/KtoContainer/utils.ts:28]: This is the same comment you made in this PR: https://github.com/Amsterdam/signals-frontend/pull/2639
But a reduce gives some typing errors, so therefore left it as is. Also a refactor seems out of scope for this PR feedback.

File: src/signals/incident/containers/KtoContainer/utils.ts [hunk 2]
```diff
@@ -58,3 +58,9 @@ export const isFetchError = (
 ): error is FetchError => {
   return (error as FetchError).detail !== undefined
 }
+
+export const stripLastCharacterIfNotLetter = (inputString: string) => {
```
[the remark this answers is not in the document]
Comment (vdegraaf) [reply src/signals/incident/containers/KtoContainer/utils.ts:62]: Will do!

**Comment by vdegraaf** [comment]
> * [ ]  why only the last letter that is being stripped? I've just entered a lot of dots, if you don't want them as the last letter, why would you want them as the second to last letter?
> * [ ]  in the email a text of 1000 characters doesn't break but continues on the same line.
> * [ ]  It is a little bit weird that you would use a inputfield for 1000 characters and an areafield for also 1000 characters (it's necessary because of the validation, but both of them on the same page look a  little weird)

1) Why would someone create a question with double question marks of dots? I think we can assume some decency form our colleagues.

2) Not sure what you mean. Can you elaborate?

3) What are you referring to? Might be better to comment in the code

**Review by OscarBakker — APPROVED** [review]

**Comment by vdegraaf** [comment]
> > * [ ]   why only the last letter that is being stripped? I've just entered a lot of dots, if you don't want them as the last letter, why would you want them as the second to last letter?
> > * [ ]   in the email a text of 1000 characters doesn't break but continues on the same line.
> > * [ ]   It is a little bit weird that you would use a inputfield for 1000 characters and an areafield for also 1000 characters (it's necessary because of the validation, but both of them on the same page look a  little weird)
> 
> 1. Why would someone create a question with double question marks of dots? I think we can assume some decency form our colleagues.
> 2. Not sure what you mean. Can you elaborate?
> 3. What are you referring to? Might be better to comment in the code
> 
> 1. Ok, but then why do we even have the function?
> 2. I'm trying to replicate. I've run into another issue, see the word doc.
>    [feedback Vincent.docx](https://github.com/Amsterdam/signals-frontend/files/12274017/feedback.Vincent.docx)
> 3. I'll make also a remark in the code, but this is what I mean.
>    image

---
1) Because I create a key-value pair based on the question so I can match the checkbox with the open question. But when there is a dot it gets stripped when it becomes a key, making the matching fail.

2) Not reproducable

3) Fixed! Made it the textArea the same sizes

**Review by SireeKoolenWijkstra — APPROVED** [review]
Very nice

**Pull request merged**